### ISSUE TITLE
list nesting

### ISSUE THREAD

**Issue by jschoch** [debois/elm-mdl#160]
Wondering what the right way to do list nesting is, it appears that my list is hidden and the docs say "no secondary content" which is confusing.

**Comment by debois** [comment]
Could you give an example of what you're trying to achieve? I'm not sure what you mean by "list nesting"?

**Comment by jschoch** [comment]
```elm
Lists.ul [] [
  Lists.li [] [
    Lists.content [] [
      Lists.ul [] [...]
      -- or,
        Html.ul [] [..]]
     ....

```

**Comment by debois** [comment]
I don't think upstream supports nested lists. The [spec](https://material.google.com/components/lists-controls.html) also doesn't mention anything about nesting lists.

Could you link to some image that looks like what you're trying to do?

**Comment by OvermindDL1** [comment]
I nested mdl lists manually in my project and am currently using them (though the sub-lists are wrapped in a div, so maybe that helps).  This was before elm-mdl had list support.

**Comment by debois** [comment]
@OvermindDL1 can you post an image? I'd like to see what that looks like.

**Comment by OvermindDL1** [comment]
See http://overminddl1.com/screenshots/work/MessengerRoomList.png
![Image](http://overminddl1.com/screenshots/work/MessengerRoomList.png)
The main list has the header of "Rooms".

It has two items in its list, the "Public Room" list and the "Room: Blah" list.

The sections "Public Room" and "Room: Blah" are headers in their respective sublists.

The rooms in the sub-sections are items in their sublists.

I have a lot of CSS styling on those but they are actually being phased out currently.

**Comment by debois** [comment]
I see. From a UI perspective, I can think of this as a "sectioned list" rather than nested lists, yes?

MDL doesn't support this; the examples in the [spec](https://material.google.com/components/lists.html#lists-specs) has it, but the spec itself doesn't really say anything about how to do it. 

I think your best option is to style the "outer" list yourself, then use `Material.List` for the inner list. Alternatively, you could do everything in one `Material.List`, then style section headers yourself (I believe this is what you're doing @OvermindDL1?). 

Are there particular options/capabilities you need to add to list items to make styling of section heads easier?

**Comment by OvermindDL1** [comment]
Yeah I have a lot of custom css.  A sectioned list would be useful, but you could potentially have more layers as well.  Could encode that as a multi-depth list of lists perhaps (although Elm missing HKT's would make that a bit more painful so probably need to wrap it up).  A tree would be quite nice as well, especially if configurable in how it is displayed.
